server/service/src/item/item_variant/upsert/validate.rs

```diff
@@ -18,11 +18,13 @@ pub fn validate(
 ) -> Result<Option<ItemVariant>, UpsertItemVariantError> {
     use UpsertItemVariantError::*;
 
-    let item = check_item_exists(connection, &input.item_id)?.ok_or(ItemDoesNotExist)?;
-
     let existing_item_variant = ItemVariantRepository::new(connection)
         .query_one(ItemVariantFilter::new().id(EqualFilter::equal_to(&input.id)))?;
 
+    if check_item_exists(connection, &input.item_id)?.is_none() {
+        return Err(UpsertItemVariantError::ItemDoesNotExist);
+    }
+
```

Comment:
Why was this check move to after the existing item variant query? 🤔 

```suggestion
   check_item_exists(connection, &input.item_id)?.ok_or(ItemDoesNotExist)?;
```

please move it back up, and `ok_or` checks if none then..